src/libawkward/array/IndexedArray.cpp

```diff
@@ -2133,6 +2133,13 @@ namespace awkward {
     util::handle_error(err2, classname(), identities_.get());
 
     ContentPtr next = content_.get()->carry(nextcarry, false);
+
+    bool inject_nones = false;
+    std::pair<bool, int64_t> branchdepth = branch_depth();
+    if (numnull > 0  &&  !branchdepth.first  &&  negaxis != branchdepth.second) {
+      keepdims = false;
```

Comment:
`keepdims` is needed for reducers, which would ordinarily remove a dimension (`keepdims` keeps the dimension from being removed, but makes its length 1). I haven't understood the role of `keepdims` for sorting, which always keeps the dimensions the same.

If it's not needed but harmless, we don't have to get rid of it right away, but should probably make a note of it.